### PR TITLE
Close #599 bookme+ email order confirmation

### LINKED ISSUE #599: BookMe+ Email Order Confirmation
### User story
- As a user of the BookMe+ app, after successfully making a booking, I want to receive an email confirmation so that I can review all the details of my booking.

### Acceptance criteria
**- The confirmation email must include the following information:**
1. Booking Details
2. Event Details
3. Customer Information
4. Important Notices
5. Customer Service 

**- The confirmation email should be sent to the email address associated with the user's account. If a user registers with a phone number, the email should be sent to the email address provided in the contact information.**

 **- The QR code is one-time checked-in, even if users have booked many tickets.**
 **- The Terms and Conditions and Privacy Policy are links that when clicked, redirect users to the BookMe.Plus web.**
**- The social media icons are clickable and redirect users to the respective social media pages.**

**- Social media's link**
1. Facebook "https://www.facebook.com/bookme.plus"
2. Instagram "https://www.instagram.com/bookme.plus"
3. Tiktok "https://www.tiktok.com/@bookme.plus"

**- Terms and condition & Privacy and Policy links (Pending)**

### Design link
https://www.figma.com/file/lozZfapNXqz2IQQkz7kQiY/Accomodation-App---Dev?type=design&node-id=4447%3A41646&mode=design&t=1ILPPnBiswNggz8H-1

email confirmation

### PR DESCRIPTION
**Pull request by kimsrung**
event mailer
hotel mailer

Fallback Background:
fallback event mailer
fallback hotel mailer

### CODE REVIEW

File: app/models/concerns/spree_cm_commissioner/product_type.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@ module SpreeCmCommissioner
   module ProductType
     extend ActiveSupport::Concern
 
-    PRODUCT_TYPES = %i[accommodation service ecommerce].freeze
+    PRODUCT_TYPES = %i[accommodation service ecommerce event].freeze
```
Comment (theachoem) [inline app/models/concerns/spree_cm_commissioner/product_type.rb:8]: For event, we can use ecommerce bong

Comment (theachoem) [inline app/models/concerns/spree_cm_commissioner/product_type.rb:8]: We can use ecommerce for event.

File: app/helpers/spree/mail_helper_decorator.rb [hunk 2]
```diff
@@ -27,6 +35,22 @@ def vendor_rating_stars(vendor)
     def format_date(day)
       day.strftime('%A %d, %B, %Y')
     end
+
+    def google_map(lat, lon)
+      "https://www.google.com/maps/@#{lat},#{lon},15z?entry=ttu"
+    end
+
+    def generate_qr(order)
+      qrcode = RQRCode::QRCode.new("https://bookme.plus/orders/#{order.number}")
```
Comment (theachoem) [inline app/helpers/spree/mail_helper_decorator.rb:44]: You can use store.url + /orders/:number instead. For dynamic, we also have staging/dev.

File: app/assets/stylesheets/spree_cm_commissioner/mailer/order/style.scss [hunk 3]
```diff
@@ -1,115 +1,188 @@
...
-      color: #333333;
-      font-weight: bold;
+  .view-map {
+    text-decoration: none;
+    color: #6444D8
+  }
+  #greeting-card {
+    color: #fff;
+    margin-top: 40px;
+    background-image: image-url("event-mailer-bg.png");
```
Comment (channainfo) [inline app/assets/stylesheets/spree_cm_commissioner/mailer/order/style.scss:50]: background image might be blocked or not supported. make sure if the background image is not there the content still renders as expected ( use a fallback background color with a similar color ). You might test to remove background image and see how it looks

**Review by channainfo — CHANGES_REQUESTED** [review]

**Comment by channainfo** [comment]
@kimsrung let's add a demo for the version without a background.

File: app/helpers/spree/mail_helper_decorator.rb (hunk 2 above)
Comment (theachoem) [reply app/helpers/spree/mail_helper_decorator.rb:44]: We can use order.store.url bong

**Comment by kimsrung** [comment]
Currently, I used only one background image for mailer. So it's already handled.

**Pull request merged**